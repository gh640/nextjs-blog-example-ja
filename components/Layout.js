import Head from "next/head"
import Link from "next/link"

const Layout = (props) => {
  const { title, children } = props
  const siteTitle = "後藤のブログ"

  return (
    <div className="page">
      <Head>
        <title>{title ? `${title} | ${siteTitle}` : siteTitle}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <header>
        <h1 className="site-title">
          <Link href="/">
            <a>{siteTitle}</a>
          </Link>
        </h1>
      </header>

      <main>
        {title ? <h1 className="page-title">{title}</h1> : ``}
        <div className="page-main">
          {children}
        </div>
      </main>

      <footer>
        &copy; {siteTitle}
      </footer>

      <style jsx>{`
        .page {
          padding: 2em 1em;
          max-width: 800px;
          margin-left: auto;
          margin-right: auto;
        }

        header {
          display: flex;
          justify-content: center;
          align-items: center;
          margin: 0 0 4em;
        }

        .site-title a {
          color: inherit;
          text-decoration: none;
        }

        footer {
          margin-top: 4em;
          padding-top: 2em;
          padding-bottom: 2em;
          display: flex;
          justify-content: center;
          align-items: center;
        }
      `}</style>

      <style jsx global>{`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: 'Noto Sans JP', -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
          color: #222;
        }

        img,
        iframe {
          max-width: 100%;
        }

        h1, h2, h3, h4, h5, h6 {
          font-family: Montserrat, -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      `}</style>
    </div>
  )
}

export default Layout
