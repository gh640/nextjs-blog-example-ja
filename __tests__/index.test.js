import Link from "next/link"
import renderer from "react-test-renderer"

import Layout from "../components/Layout"
import Home from "../pages/index.js"

jest.mock("next/link", () => {
	return ({ children, ...attrs }) => {
		return <a {...attrs}>{children}</a>
	}
})
jest.mock("../components/Layout")

describe("Home", () => {
	Layout.mockImplementation(({ children }) => children)

	it("renders correctly without posts nor archive", () => {
		const tree = renderer
			.create(<Home posts={[]} hasArchive={false}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders correctly with posts and archive", () => {
		const posts = [
			{
				title: "かきくけこ",
				slug: "kakikukeko",
				published: "2020/08/27",
			},
			{
				title: "あいうえお",
				slug: "aiueo",
				published: "2018/05/27",
			},
		]
		const tree = renderer
			.create(<Home posts={posts} hasArchive={true}/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
