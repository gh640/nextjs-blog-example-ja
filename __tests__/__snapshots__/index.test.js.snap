// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly with posts and archive 1`] = `
Array [
  <div
    className="jsx-4006167202 post-teaser"
  >
    <h2
      className="jsx-4006167202"
    >
      <a
        className="jsx-4006167202"
        href="/posts/[id]"
      >
        かきくけこ
      </a>
    </h2>
    <div
      className="jsx-4006167202"
    >
      <span
        className="jsx-4006167202"
      >
        2020/08/27
      </span>
    </div>
  </div>,
  <div
    className="jsx-4006167202 post-teaser"
  >
    <h2
      className="jsx-4006167202"
    >
      <a
        className="jsx-4006167202"
        href="/posts/[id]"
      >
        あいうえお
      </a>
    </h2>
    <div
      className="jsx-4006167202"
    >
      <span
        className="jsx-4006167202"
      >
        2018/05/27
      </span>
    </div>
  </div>,
  <div
    className="jsx-4006167202 home-archive"
  >
    <a
      className="jsx-4006167202"
      href="/archive/[page]"
    >
      アーカイブ
    </a>
  </div>,
]
`;

exports[`Home renders correctly without posts nor archive 1`] = `""`;
